packages/synapse-react-client/src/components/SynapseTable/datasets/DatasetItemsEditor.tsx

```diff
@@ -172,25 +244,24 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
   } = useMemo(() => getCopy(fetchedDataset), [fetchedDataset])
 
   useEffect(() => {
-    // Don't update when we already have datasetToUpdate
-    if (!datasetToUpdate && fetchedDataset) {
+    if (fetchedDataset) {
```

Comment:
`fetchedDataset` should only change when we really should re-sync with the server